Add Support for Gitalk & Fix something

Fidelxyz opened the pull request:
新增：

- 添加了对 [Gitalk](https://github.com/gitalk/gitalk) 的支持，支持 lazyload。

  为了避免 Gitalk 与 Valine 冲突，我对涉及到 Valine 的 `comment.njk` 和 `pjax.js` 做了一些修改，但我现在没有使用 Valine 不方便测试，**建议合并前测试一下 Valine 的可用性**。

修复 / 优化：

- 修复 `<root>/_config.yml` 中 `description` 为空时 `pangu` 报错 `spacing(text) only accepts string but got object` 的问题。
- Polyfill.js 换为 v3 版本，引用了带有 `.min` 后缀的压缩资源。#8
  更换为 Alicdn 的镜像 <https://polyfill.alicdn.com/polyfill.min.js>。(Suggested by @SerokSSR)
- ~~Google Fonts 源换为了 [loli.net 的反代加速资源](https://sb.sb/blog/css-cdn/)。~~
  更换为基于 Tencent CDN 的镜像源 <https://fonts.dogedoge.com>。(Suggested by @SerokSSR)
- 注释掉 `<theme>/_config.yml` 的部分配置，以便不需要这些功能的用户不需要修改 `<theme>/_config.yml`。需要这些功能的用户可自行在 `<root>/_config.shoka.yml` 中配置。

_（另外大佬的主题真的太好看了！）_

Commit:
Add Support for Gitalk & Fix something

Fidelxyz commented on layout/_macro/segment.njk:
```diff
@@ -3,10 +3,12 @@
   {%- set postTitleIcon = '<i class="ic i-link-alt"></i>' %}
 {%- endif %}
 {%- set postText = item.title or item.link or __('post.untitled') %}
-<article class="item">
```
The change in this file should be removed...
I submitted it by mistake. :(

SerokSSR commented:
Alicdn now provides a Polyfill mirror https://polyfill.alicdn.com/polyfill.min.js, which is much faster.

As for Google Fonts, loli.net has moved its server to HK(which obviously lead to a decrease in speed) since its ICP license has been cancelled. Another solution is to use fonts.dogedoge.com, based on Tencent CDN.

Commit:
Update

- Fix segment.njk
- Use a Polyfill mirror provided by Alicdn
- Use a font mirror fonts.dogedoge.com based on Tencent CDN

Fidelxyz commented:
@SerokSSR Thanks! I've submitted these changes in the latest commit.

Besides, the problem of `segment.njk` caused by my mistake has been fixed.

Commit:
Update segment.njk

Commit:
Fix incompatibility with Hexo 5.4

Fidelxyz committed:
Fix bug (gitalk and archine & tag pages)
